modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase

```diff
@@ -890,6 +890,80 @@ definition {
 		}
 	}
 
+	@description = "This is a test case for COMMERCE-11003. Assert only the product options that are added to SKUs can be displayed in the product detail widget"
+	@priority = 5
+	test CanDisplayOptionsOnlyAddedToProductSKUs {
+		property portal.acceptance = "true";
+
+		task ("Given a Minium site") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("When creating a simple product with four values in an option and enabling SKU Contributor for that option") {
+			CommerceProducts.openProductsAdmin();
+
+			CPCommerceCatalog.newProductsEntry(
+				productCatalog = "Minium",
+				productName = "Test",
+				productType = "Simple");
```

Comment:
The creation of a product can be done using the macro `CommerceJSONProductsAPI._addCommerceProduct()`. Please check the code to see how it's used.